src/na/na_types.h

```diff
@@ -31,10 +31,12 @@ typedef na_uint64_t na_offset_t;               /* Offset */
 
 /* Init info */
 struct na_init_info {
-    const char *ip_subnet;     /* Preferred IP subnet */
-    const char *auth_key;      /* Authorization key */
-    na_uint32_t progress_mode; /* Progress mode */
-    na_uint8_t max_contexts;   /* Max contexts */
+    const char *ip_subnet;         /* Preferred IP subnet */
+    const char *auth_key;          /* Authorization key */
+    na_size_t max_unexpected_size; /* Max unexpected size hint */
+    na_size_t max_expected_size;   /* Max expected size hint */
```

Comment:
Added those 2 new flags. These are just hints provided by users and NA plugins do not have to honor those limits if they can't / don't want to support it.